src/Http/Http/test/StreamPipeReaderTests.cs

```diff
@@ -477,6 +477,42 @@ public async Task ConsumePartialBufferBetweenMultipleSegmentsWorks()
             Reader.AdvanceTo(readResult.Buffer.End);
         }
 
+        [Fact]
+        public async Task SetMinimumReadThresholdSegmentAdvancesCorrectly()
+        {
+            Reader = new StreamPipeReader(MemoryStream,
+                minimumSegmentSize: 16,
+                minimumReadThreshold: 8,
+                new TestMemoryPool());
+
+            Write(Encoding.ASCII.GetBytes(new string('a', 8)));
+            var readResult = await Reader.ReadAsync();
+            Reader.AdvanceTo(readResult.Buffer.Start, readResult.Buffer.End);
+
+            Write(Encoding.ASCII.GetBytes(new string('a', 16)));
+            readResult = await Reader.ReadAsync();
+
+            Assert.False(readResult.Buffer.IsSingleSegment);
```

Comment:
Assert buffer segment sizes in both of these tests.

Reply:
Because as far as I understand this test would pass even without your change.

Reply:
It would fail. The second call to ReadAsync would fill the original segment rather filling a new segment, meaning there a more than 2 segments in this case.